cats/cats-redis/src/test/java/com/netflix/spinnaker/cats/redis/cluster/ClusteredSortAgentSchedulerTest.java

```diff
@@ -0,0 +1,137 @@
+ */
+
+package com.netflix.spinnaker.cats.redis.cluster;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.netflix.spinnaker.cats.agent.CachingAgent;
+import com.netflix.spinnaker.cats.agent.DefaultCacheResult;
+import com.netflix.spinnaker.cats.agent.ExecutionInstrumentation;
+import com.netflix.spinnaker.cats.cluster.DefaultAgentIntervalProvider;
+import com.netflix.spinnaker.cats.test.TestAgent;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.commons.lang3.reflect.MethodUtils;
+import org.junit.Before;
+import org.junit.Test;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+
+public class ClusteredSortAgentSchedulerTest {
+
+  private ClusteredSortAgentScheduler clusteredSortAgentScheduler;
+
+  private Jedis jedis = mock(Jedis.class);
+  private JedisPool jedisPool = mock(JedisPool.class);
+  private Integer parallelism = 2;
+  private CachingAgent.CacheExecution agentExecution = mock(CachingAgent.CacheExecution.class);
+  private ExecutionInstrumentation executionInstrumentation = mock(ExecutionInstrumentation.class);
+
+  private String WAITING_SET;
+  private String WORKING_SET;
+  private Method saturatePoolMethod;
+  private Optional<Semaphore> runningAgents;
+
+  @Before
+  public void setUp() throws IllegalAccessException {
+    when(jedisPool.getResource()).thenReturn(jedis);
+    when(jedis.scriptLoad(anyString())).thenReturn("testScriptSha");
+    when(jedis.time()).thenReturn(List.of("1678784468", "374338"));
+    DefaultAgentIntervalProvider intervalProvider = new DefaultAgentIntervalProvider(6000000);
+    clusteredSortAgentScheduler =
+        new ClusteredSortAgentScheduler(jedisPool, () -> false, intervalProvider, parallelism);
+
+    WAITING_SET =
+        (String)
+            FieldUtils.getDeclaredField(ClusteredSortAgentScheduler.class, "WAITING_SET", true)
+                .get(clusteredSortAgentScheduler);
+    WORKING_SET =
+        (String)
+            FieldUtils.getDeclaredField(ClusteredSortAgentScheduler.class, "WORKING_SET", true)
+                .get(clusteredSortAgentScheduler);
+    saturatePoolMethod =
+        MethodUtils.getMatchingMethod(ClusteredSortAgentScheduler.class, "saturatePool");
+    saturatePoolMethod.setAccessible(true);
+    runningAgents =
```

Comment:
I'd say it's worth adding a getter for testing, but it's also nice to leave it hidden.

Reply:
Yeah, i prefer to leave it hidden.